### PR TITLE
Fixes #18513 - Redraw job invocation status graph properly

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: app/helpers/remote_execution_helper.rb [hunk 1]
```diff
@@ -12,30 +12,34 @@ def job_invocation_chart(invocation)
...
     if (bulk_task = invocation.task)
-      failed_tasks = bulk_task.sub_tasks.select { |sub_task| task_failed? sub_task }
-      cancelled_tasks, failed_tasks = failed_tasks.partition { |task| task_cancelled? task }
-      success = bulk_task.output['success_count'] || 0
-      cancelled = cancelled_tasks.length
-      failed = failed_tasks.length
-      pending = (bulk_task.output['pending_count'] || bulk_task.sub_tasks.count)
+      unless invocation.queued?
+        total = invocation.targeting.resolved? ? invocation.targeting.hosts.count : 0
+        failed, not_failed = bulk_task.sub_tasks.partition { |sub_task| task_failed? sub_task }
```
Comment (iNecas) [inline app/helpers/remote_execution_helper.rb]: Keep in mind there can be several thousand of the tasks in remote execution job for larger infrastructures: it's better to prefer sql queries here

Comment (iNecas) [inline app/helpers/remote_execution_helper.rb]: Also, I would suggest putting the data gathering into JobInvocation model and write test for this.

**Comment by adamruzicka** [comment]
Now requires https://github.com/theforeman/foreman-tasks/pull/236

**Comment by ohadlevy** [comment]
I know its late in the process, but would you consider using the new C3 based charts instead of the legacy no longer maintained flot ?

**Comment by iNecas** [comment]
@ohadlevy not in this PR

**Comment by adamruzicka** [comment]
[test]

**Comment by iNecas** [comment]
Thanks @adamruzicka

**Pull request merged**